app/src/main/java/org/schabi/newpipe/settings/NotificationsSettingsFragment.kt

```diff
@@ -0,0 +1,111 @@
+    }
+
+    override fun onStop() {
+        defaultPreferences.unregisterOnSharedPreferenceChangeListener(this)
+        super.onStop()
+    }
+
+    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences?, key: String?) {
+        val context = context ?: return
+        if (key == getString(R.string.streams_notifications_interval_key) || key == getString(R.string.streams_notifications_network_key)) {
+            NotificationWorker.schedule(context, ScheduleOptions.from(context), true)
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        val enabled = NotificationHelper.isNotificationsEnabledNative(context)
+        preferenceScreen.isEnabled = enabled
+        if (!enabled) {
+            if (notificationWarningSnackbar == null) {
+                notificationWarningSnackbar = Snackbar.make(
+                    listView,
+                    R.string.notifications_disabled,
+                    Snackbar.LENGTH_INDEFINITE
+                ).apply {
+                    setAction(R.string.settings) { v ->
+                        NotificationHelper.openNativeSettingsScreen(v.context)
+                    }
+                    setActionTextColor(
+                        ContextCompat.getColor(
+                            requireContext(),
+                            R.color.snackbar_action_color
+                        )
+                    )
+                    addCallback(object : Snackbar.Callback() {
+                        override fun onDismissed(transientBottomBar: Snackbar, event: Int) {
+                            super.onDismissed(transientBottomBar, event)
+                            notificationWarningSnackbar = null
+                        }
+                    })
+                    show()
+                }
+            }
+        } else {
+            notificationWarningSnackbar?.dismiss()
+            notificationWarningSnackbar = null
+        }
+        loader?.dispose()
+        loader = SubscriptionManager(requireContext())
+            .subscriptions()
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(this::updateSubscriptions, this::onError)
+    }
+
+    override fun onPause() {
+        loader?.dispose()
+        loader = null
+        super.onPause()
+    }
+
+    private fun updateSubscriptions(subscriptions: List<SubscriptionEntity>) {
+        var notified = 0
+        for (subscription in subscriptions) {
+            if (subscription.notificationMode != NotificationMode.DISABLED) {
+                notified++
+            }
+        }
+        val preference = findPreference<Preference>(getString(R.string.streams_notifications_channels_key))
+        if (preference != null) {
+            preference.summary = String.format(Locale.US, "%d/%d", notified, subscriptions.size)
+        }
+    }
+
+    private fun onError(e: Throwable) {
+        if (BuildConfig.DEBUG) {
+            e.printStackTrace()
+        }
+    }
```

Comment:
Shouldn't we notify the user about the error?

Reply:
Done